docs/reference/builder.md

```diff
@@ -1045,6 +1045,9 @@ For example:
 The output of the final `pwd` command in this `Dockerfile` would be
 `/path/$DIRNAME`
 
+Each `WORKDIR` will be created even if its not used in any subsequent `Dockerfile`
+instruction.
```

Comment:
Perhaps include it in the first paragraph; https://github.com/docker/docker/pull/21340/files#diff-4fc21624aa5dadd153e2780910da7c7dR1023, e.g. "If the workdir doesn't exist, ..."?